pkg/operator/reports.go

```diff
@@ -734,6 +740,65 @@ func (op *Reporting) runReport(logger log.FieldLogger, report *metering.Report)
 	return nil
 }
 
+func deleteReport(logger log.FieldLogger, report *metering.Report, op *Reporting) {
+	if reportNotUsedAsInput(logger, report, op) {
+		logger.Infof("Deleted report after retention period: %s, past reportingEnd: %s",
+			report.Spec.Expiration, report.Spec.ReportingEnd)
+	}
+}
+
+func reportNotUsedAsInput(logger log.FieldLogger, report *metering.Report, op *Reporting) bool {
```

Comment:
Same here, I would assume we would return a boolean, declaring whether or not the `@report` has input dependencies and any error that may occur during the process of determining the value of that return type.

Reply:
same answer as above